src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -466,12 +499,14 @@ WorkerCreateShard(Oid relationId, char *nodeName, uint32 nodePort,
 							 escapedDDLCommand);
 		}
 
-		queryResultList = ExecuteRemoteQuery(nodeName, nodePort, newShardOwner,
-											 applyDDLCommand);
-		if (queryResultList == NIL)
+		if (createInSeparateTransaction)
 		{
-			shardCreated = false;
-			break;
+			ExecuteRemoteQuery(nodeName, nodePort, newShardOwner,
```

Comment:
We'd rather avoid calls to `ExecuteRemoteQuery`. If you open a new connection (using `FORCE_NEW_CONNECTION`) and don't call `RemoteTransactionBeginIfNecessary`, then the statements be executed outside of a transaction block if you just do `ExecuteCriticalRemoteCommand`.